### PR TITLE
Add DisableTransitiveProjectReferences

### LINKED ISSUE #30341: Add DisableTransitiveProjectReferences
Recently i've found this property to have a use-case in which you don't want transitive dependencies to be visible to the compiler for your respective project. This also has synergy with directory.build.props files to do it in a global fashion for all .csprojs.

A typical solution might look like the following:

```
<DisableTransitiveProjectReferences>true</DisableTransitiveProjectReferences>
```

and

```
      <Private>false</Private>
      <ExcludeAssets>all</ExcludeAssets>
```

This feature being searchable on .NET docs would be useful for those who prefer the "old non-SDK .csproj" behavior.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cb4428fc-215c-78e1-5ff0-fe73f7a08260
* Version Independent ID: 6741b1de-38a4-fbb1-d79d-acc47d549a2d
* Content: [MSBuild properties for Microsoft.NET.Sdk - .NET](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props)
* Content Source: [docs/core/project-sdk/msbuild-props.md](https://github.com/dotnet/docs/blob/main/docs/core/project-sdk/msbuild-props.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by JonDouglas**
Fixes #30341

## Summary

Adds a few paragraphs and documentation links to `DisableTransitiveProjectReferences`.

### CODE REVIEW

File: docs/core/project-sdk/msbuild-props.md [hunk 1]
```diff
@@ -964,6 +965,20 @@ Set this property to `true` to disable implicit `FrameworkReference` or [Package
 </PropertyGroup>
 ```
 
+### DisableTransitiveProjectReferences
+
+The `DisableTransitiveProjectReferences` property will prevent implicit project references giving you similar non-transitive behavior to the [legacy project system](https://github.com/dotnet/project-system/blob/main/docs/feature-comparison.md).
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:970]:
```suggestion
The `DisableTransitiveProjectReferences` property controls implicit project references. Set this property to `true` to disable implicit `ProjectReference` items. Disabling implicit project references results in non-transitive behavior similar to the [legacy project system](https://github.com/dotnet/project-system/blob/main/docs/feature-comparison.md).
```

File: docs/core/project-sdk/msbuild-props.md [hunk 2]
```diff
@@ -964,6 +965,20 @@ Set this property to `true` to disable implicit `FrameworkReference` or [Package
...
+
+This has a similar effect to that of [`PrivateAssets="All"`](https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets) on all of the dependencies of the depended-upon project.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:972]:
```suggestion
When this property is `true`, it has a similar effect to that of setting [`PrivateAssets="All"`](/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets) on all of the dependencies of the depended-upon project.
```

File: docs/core/project-sdk/msbuild-props.md [hunk 3]
```diff
@@ -964,6 +965,20 @@ Set this property to `true` to disable implicit `FrameworkReference` or [Package
...
+
+Set this property to `true` to disable implicit `ProjectReference` items. If you set this property to `true`, you can add explicit references to just the projects you need.
```
Comment (gewarren) [inline docs/core/project-sdk/msbuild-props.md:974]:
```suggestion
If you set this property to `true`, you can add explicit references to just the projects you need.
```

**Review by gewarren — APPROVED** [review]
I moved some stuff around in my suggestions.

**Comment by JonDouglas** [comment]
@gewarren do we need to re-run the build for the validation warnings? Doesn't seem to match.

**Pull request merged**